### PR TITLE
[SLUGS] misc fixes

### LINKED ISSUE #1273: REST API URL changes when 'Slugs translations(Experimental)' turned on, causing plugin issues
When 'Slugs translations(Experimental)' is turned on in qTranslate XT, the REST API URL for Wordpress changes from 'wordpress.org/wp-json/' to 'wordpress.org/en/wp-json/', which causes issues with certain plugins. For example, Burst Statistics has been reported to have problems when this feature is enabled.

Unfortunately, we are unable to consistently reproduce this issue. However, @sviluppomania has reported experiencing this problem.

We believe that this issue could be resolved by disabling qTranslate from changing the REST API URLs. We would appreciate any assistance in implementing this solution or any other suggestions for resolving the issue. Thank you.

### LINKED ISSUE #1358: [Slug translation module] Preview a draft post doesn't work
**Describe the bug**
Previewing a draft post throws error. Tested with a custom post type (name: project, languages: de (projekt), en
 (project)).

`Stack trace:
#0 /home/webi4se86/html/wp-content/plugins/qtranslate-xt/src/modules/slugs/slugs.php(438): QTX_Module_Slugs->get_page_by_path(NULL, 'project')
#1 /home/webi4se86/html/wp-includes/class-wp-hook.php(308): QTX_Module_Slugs->filter_request(Array)
#2 /home/webi4se86/html/wp-includes/plugin.php(205): WP_Hook->apply_filters(Array, Array)
#3 /home/webi4se86/html/wp-includes/class-wp.php(390): apply_filters('request', Array)
#4 /home/webi4se86/html/wp-includes/class-wp.php(780): WP->parse_request('')
#5 /home/webi4se86/html/wp-includes/functions.php(1334): WP->main('')
#6 /home/webi4se86/html/wp-blog-header.php(16): wp()
#7 /home/webi4se86/html/index.php(17): require('/home/webi4se86...')
#8 {main}
  thrown in /home/webi4se86/html/wp-content/plugins/qtranslate-xt/src/modules/slugs/slugs.php on line 1039`

**To Reproduce**
1. Go to admin post list page
2. Click the preview link of a post with status draft
3. See error

**Expected behavior**
The post should be shown.

**Debug info**
Versions

```
{
  "PHP_VERSION": "8.2.6",
  "WP_VERSION": "6.2.2",
  "QTX_VERSION": "3.15.2",
  "Plugins": [
    "ACF Image Sizes 1.0.0",
    "Adminimize 1.11.9",
    "Advanced Custom Fields PRO 5.8.8",
    "Classic Editor 1.6.3",
    "Custom Post Type UI 1.13.7",
    "Disable Comments 2.4.4",
    "Email Address Encoder 1.0.22",
    "Loco Translate 2.6.4",
    "Post Types Order 2.0.9",
    "qTranslate-XT 3.15.2"
  ]
}
```

Configuration

```
{
  "default_language": "de",
  "enabled_languages": [
    "de",
    "en"
  ],
  "locale": {
    "de": "de_DE",
    "en": "en_US"
  },
  "date_format": {
    "de": "%A, \\d\\e\\r %e. %B %Y",
    "en": "%A %B %e%q, %Y"
  },
  "time_format": {
    "de": "%H:%M",
    "en": "%I:%M %p"
  },
  "url_mode": 2,
  "use_strftime": 3,
  "filter_options_mode": 0,
  "language_name_case": 0,
  "detect_browser_language": true,
  "hide_untranslated": false,
  "show_menu_alternative_language": false,
  "show_displayed_language_prefix": false,
  "show_alternative_content": true,
  "hide_default_language": false,
  "use_secure_cookie": false,
  "header_css_on": true,
  "text_field_filters": [],
  "disable_client_cookies": false,
  "url_info": {
    "cookie_lang_front": "de",
    "cookie_lang_admin": "en",
    "cookie_front_or_admin_found": true,
    "scheme": "https",
    "host": "www.elstertainment.de",
    "path": "/wp-admin/admin-ajax.php",
    "query": "action=admin_debug_info",
    "path-base": "",
    "wp-path": "/wp-admin/admin-ajax.php",
    "language_neutral_path": true,
    "http_referer": "https://www.elstertainment.de/wp-admin/options-general.php?page=qtranslate-xt",
    "referer_admin": true,
    "doing_front_end": false,
    "lang_cookie_admin": "en",
    "lang_admin": "en",
    "language": "en",
    "set_cookie": false
  },
  "language": "en",
  "editor_mode": 0,
  "highlight_mode": 1,
  "auto_update_mo": true,
  "hide_lsb_copy_content": false,
  "lsb_style": "simple-buttons.css",
  "config_files": [
    "./i18n-config.json"
  ],
  "custom_fields": [],
  "custom_field_classes": [],
  "post_type_excluded": [
    "acf-field-group",
    "acf-field"
  ],
  "admin_enabled_modules": {
    "acf": true,
    "slugs": true,
    "all-in-one-seo-pack": false,
    "events-made-easy": false,
    "jetpack": false,
    "google-site-kit": false,
    "gravity-forms": false,
    "woo-commerce": false,
    "wp-seo": false
  },
  "translator": {},
  "qtrans_compatibility": false
}
```

Browser

```
{
  "cookies": [],
  "navigator": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36",
  "Javascript built-in RegExp: @@split": "supported"
}
```

### LINKED ISSUE #1359: [Slug translation module] hreflang has wrong value of Locale at front-end
When Slugs active and "Locale at front-end" have value in Language Editor, hreflang is set "Language Code" and ignore "Locale at front-end". When Slugs deactivated hreflang is set from value of "Locale at front-end". Some small bug. 

1. Go to page wp-admin/options-general.php?page=qtranslate-xt&edit=ua
2. Edit value 'Locale at front-end' to "uk"
3. Click Save
4. Go to page wp-admin/options-general.php?page=qtranslate-xt#integration
5. Slugs translation (activate / deactivate)
6. See in code of page (<link hreflang="uk" if deactivated Slugs, <link hreflang="ua" if activated Slugs )

**Expected behavior**
On all states of Slugs, hreflang have value "Locale at front-end" if is set.
```
qTranslate-XT  3.15.2
```
Thanks =)

### LINKED ISSUE #1373: Error with Slug active
Greetings,
I use the All In One WP Security Plugin and among these I also use the anti-brute force setting by renaming the access URL, never had any problems between Qtranslate-XT and All In One WP Security, but now if Integration remains active Slugs translation it causes me error when accessing the login page with custom url, the following error:

```
**PHP Fatal error: Uncaught TypeError:** QTX_Module_Slugs::home_url(): Argument #2 ($path) must be of type string, null given, called in ./.../.../wp-includes/class-wp-hook.php on line 310 and defined in ./.../.../wp-content/plugins/qtranslate-xt/src/modules/slugs/slugs.php:531
--
Stack trace:
#0 ./.../.../wp-includes/class-wp-hook.php(310): QTX_Module_Slugs->home_url()
#1 ./.../.../wp-includes/wp-includes/plugin.php(205): WP_Hook->apply_filters()
#2 ./.../.../wp-includes/wp-includes/link-template.php(3456): apply_filters()
#3 ./.../.../wp-includes/wp-includes/link-template.php(3402): get_home_url()
#4 ./.../.../wp-includes/wp-content/plugins/oauth2-provider/library/class-wo-api.php(44): home_url()
#5 ./.../.../wp-includes/wp-content/plugins/oauth2-provider/wp-oauth-server.php(112): include_once('./.../...')
#6 ./.../.../wp-includes/wp-includes/class-wp-hook.php(310): wpoauth_server_template_redirect_intercept()
#7 ./.../.../wp-includes/wp-includes/plugin.php(205): WP_Hook->apply_filters()
#8 ./.../.../wp-includes/wp-includes/template-loader.php(104): apply_filters()
#9 ./.../.../wp-includes/wp-blog-header.php(19): require_once('./.../...')
#10 ./.../index.php(17): require('./.../...')
#11 {main}
thrown in ./.../.../wp-content/plugins/qtranslate-xt/src/modules/slugs/slugs.php on line 531
```

It should be noted that in addition to this error I had another similar one with the same integration but with another "Burst Statistics" plugin which obscures the statistics and this appears: Server error: The response is not a valid JSON response.

The appearance of errors is not caused by the Qtranslate-XT Plugin itself, but solely by the SLUG integration (if active).

**Debug info**:
"PHP_VERSION": "8.2.8",
"WP_VERSION": "6.3.1",
"QTX_VERSION": "3.15.2",

### PR DESCRIPTION
**Pull request by spleen1981**
Bunch of fixes for slugs module:
- Slugs: fix possible null passed to filter_request functions
Fixes #1358 

- Slugs: filter_request() reformatting and cleanup

- Slugs: use 'Locale at front-end' if available as hreflang
Fixes #1359 

- Slugs: remove reference to obsolete hook

- Slugs: fix home_url to handle 'rest' and 'relative' schemes
This is needed to fix wrong result in case of 'relative' scheme argument request, being currently returned the entire 'http/https' anyways.
Fixes #1373 
Fixes #1273

- Slugs: restore original query if filter_request() fails
In case filter_request is not able to identify an handling function, original query needs to be provided (including the 404 flag previously stripped)

### CODE REVIEW

**Comment by spleen1981** [comment]
@herrvigg better "Rebase and merge" for this PR in case, for cleaner history.

**Comment by sviluppomania** [comment]
HI,

this solution resolves 2 reported BUGS:

#1273
#1373

Thank you.

File: src/modules/slugs/slugs.php [hunk 1]
```diff
@@ -72,22 +70,29 @@ function init() {
...
-            echo '<link hreflang="' . $language . '" href="' . esc_url( $this->get_current_url( $language ) ) . '" rel="alternate" />' . "\n";
+        $hreflangs = array();
+        foreach ( $q_config['enabled_languages'] as $lang ) {
+            $hreflang = ! empty ( $q_config['locale_html'][ $lang ] ) ? $q_config['locale_html'][ $lang ] : $lang;
+            $hreflangs[ $hreflang ] = esc_url( $this->get_current_url( $lang ) );
+        }
+        $hreflangs['x-default'] = esc_url( $this->get_current_url( $q_config['default_language'] ) );
+
+        foreach ( $hreflangs as $hreflang => $href ) {
+            echo '<link hreflang="' . $hreflang . '" href="' . $href . '" rel="alternate" />' . PHP_EOL;
```
Comment (herrvigg) [inline src/modules/slugs/slugs.php:95]: Not blocking for this patch: we have duplicated code and logic coming from 
https://github.com/qtranslate/qtranslate-xt/blob/7c89d2514f9833184e70322a292d149022736fe5/src/frontend.php#L44

The two should always remain consistent.

**Comment by herrvigg** [comment]
Great with a lot of fixes, but the review and follow-up is made difficult by "squashing" all in a single commit. Since it's very different topics I would prefer separate fixes and merge them one by one.
Let's go for a merge.

**Pull request merged**